packages/fleet/lib/node-providers/node_provider.ts

```diff
@@ -7,4 +7,6 @@ export interface NodeProvider {
     start(node: Node): Promise<Result<void>>;
     terminate(node: Node): Promise<Result<void>>;
     verifyUrl(url: string): Promise<Result<void>>;
+    finish(node: Node): Promise<Result<void>>;
```

Comment:
The way nodes are `finished` was specific to instance-based nodes (i.e. Render compute instances or Kubernetes pods). Offloading this capability to the node provider allows us to have custom functionality for different kinds of nodes.